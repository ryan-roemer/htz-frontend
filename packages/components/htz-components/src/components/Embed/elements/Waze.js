import React from 'react';
import PropTypes from 'prop-types';
import { createComponent, } from 'react-fela';
import Caption from '../Caption';

Waze.propTypes = {
  inputTemplate: PropTypes.string.isRequired,
  embedType: PropTypes.string.isRequired,
  caption: PropTypes.string,
  credit: PropTypes.string,
  settings: PropTypes.shape({
    pin: PropTypes.bool,
    coordinates: PropTypes.array.isRequired,
    language: PropTypes.string.isRequired,
  }).isRequired,
};

Waze.defaultProps = {
  caption: '',
  credit: '',
};

const mapWrapper = ({ aspectRatio, nyt, }) => {
  const [ width, height, ] = aspectRatio ? aspectRatio.split('/') : [ 16, 9, ];
  const aspect = `${(height / width) * 100}%`;
  const paddingTop = nyt ? '69px' : '';

  return {
    margin: '0',
    paddingBottom: aspect,
    height: '0',
    overflow: 'hidden',
    position: 'relative',
    paddingTop,
  };
};

const MapWrapper = createComponent(mapWrapper, 'figure');

const mapElement = () => ({
  margin: '0',
  padding: '0',
  height: '100% !important',
  width: '100% !important',
  left: '0',
  top: '0',
  position: 'absolute',
  display: 'block',
  border: 'none',
});

const MapElement = createComponent(mapElement, 'iframe', props =>
  Object.keys(props)
);

function Waze(props) {
  const settings = props.settings;
  const pin = settings.pin ? '&pin=1' : '';
  const language = settings.language;
  const coordinates = settings.coordinates;

  return (
    <div>
      <MapWrapper>
        <MapElement
          width="600"
          height="450"
          src={`https://embed.waze.com/${language}/iframe${coordinates[0]}&${coordinates[1]}&${coordinates[2]}&${pin}`}
          frameBorder="0"
          allowFullScreen=""
        />
      </MapWrapper>
      <Caption
        caption={props.caption}
        credit={props.credit}
        inputTemplate={props.inputTemplate}
        embedType={props.embedType}
      />
    </div>
  );
}

export default Waze;
