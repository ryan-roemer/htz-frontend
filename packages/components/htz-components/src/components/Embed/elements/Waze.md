<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
**Table of Contents**

- [Waze component](#waze-component)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

### Waze component

An example for Waze map.

```jsx static
<Waze
  settings = {{
    pin: 'true',
    coordinates: [
      '?zoom=15',
      'lat=32.06408',
      'lon=34.77688',
    ],
    language: 'he',
  }}
  caption = "traffic map"
  credit = "Waze"
/>
```
