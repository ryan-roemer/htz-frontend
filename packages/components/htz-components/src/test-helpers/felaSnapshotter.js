import React from 'react';
import createRenderer from '../components/StyleProvider/createRenderer';
import StyleProvider from '../components/StyleProvider/StyleProvider';
import createFelaSnapshotter from './createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: true, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer);

export default felaSnapshotter;
