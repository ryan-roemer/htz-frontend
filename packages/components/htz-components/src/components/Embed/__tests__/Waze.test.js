import React from 'react';
import felaSnapshotter from '../../../test-helpers/felaSnapshotter';

import Waze from '../elements/Waze';

it('Waze ', () => {
  const snapshot = felaSnapshotter(
    <Waze
      content="https://www.waze.com/livemap?zoom=15&lat=32.06408&lon=34.77688"
      caption="Waze"
      credit="Waze"
      embedType="waze"
      elementType="embedElement"
      settings={{
        pin: 'true',
        coordinates: [ '?zoom=15', 'lat=32.06408', 'lon=34.77688', ],
        language: 'he',
      }}
      inputTemplate="com.polobase.WazeEmbed"
      contentId="7.5003"
      contentName="Waze"
    />
  );
  expect(snapshot).toMatchSnapshot();
});
