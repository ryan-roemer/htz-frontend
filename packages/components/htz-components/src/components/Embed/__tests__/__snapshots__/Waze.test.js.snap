// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Waze  1`] = `
Object {
  "component": <div>
    <figure
      className="htz-a htz-b htz-c htz-d htz-e htz-f"
    >
      <iframe
        allowFullScreen=""
        className="htz-a htz-g htz-h htz-i htz-j htz-k htz-l htz-m htz-n"
        frameBorder="0"
        height="450"
        onLoad={null}
        src="https://embed.waze.com/he/iframe?zoom=15&lat=32.06408&lon=34.77688&&pin=1"
        width="600"
      />
    </figure>
  </div>,
  "styles": ".htz-a{margin:0rem}.htz-b{padding-bottom:56.25%}.htz-c{height:0rem}.htz-d{overflow:hidden}.htz-e{position:relative}.htz-f{padding-top:}.htz-g{padding:0rem}.htz-h{height:100% !important}.htz-i{width:100% !important}.htz-j{left:0rem}.htz-k{top:0rem}.htz-l{position:absolute}.htz-m{display:block}.htz-n{border:none}",
}
`;
